### PR TITLE
feat: [SKU modularization] AWS chart changes

### PR DESCRIPTION
**Pull request by smritidahal653**
**Reason for Change**:

Updating kaito charts for aws integration:

- add clusterrole permission for ec2nodeclasses resource
- add new env variable in deployment CLUSTER_NAME as this environment is needed to create EC2 Node Class
- update nodeSelectorTerm for nvidia-device-plugin daemonset based on the cloud provider
- add AWS specific nodeSelectorRequirement for nodeClaim obj
- add InstanceStorePolicyRAID0 to EC2 node class to share node's ephermeral storage among pods that request it. Additional information [here](https://karpenter.sh/v1.0/concepts/nodeclasses/#specinstancestorepolicy)
- fixing a bug on nodeClaim create (kubeClient.Create) - if the nodeClaim fails to get created, we retry to create it again. However, on kubeClient.Create, certain metadata are set like resourceVersion which will be passed along in the retires if we don't pass the DeepCopy(). This metadata being populated will throw an error from KubeClient `resourceVersion should not be set on objects to be created`

**Requirements**

- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: pkg/utils/nodeclaim/nodeclaim_test.go [hunk 1]
```diff
@@ -205,7 +205,7 @@ func TestGenerateNodeClaimManifest(t *testing.T) {
 		assert.Equal(t, nodeClaim.Labels[kaitov1alpha1.LabelWorkspaceNamespace], mockWorkspace.Namespace, "label must have same workspace namespace as workspace")
 		assert.Equal(t, nodeClaim.Labels[consts.LabelNodePool], consts.KaitoNodePoolName, "label must have same labels as workspace label selector")
 		assert.Equal(t, nodeClaim.Annotations[v1beta1.DoNotDisruptAnnotationKey], "true", "label must have do not disrupt annotation")
-		assert.Equal(t, len(nodeClaim.Spec.Requirements), 3, " NodeClaim must have 3 NodeSelector Requirements")
+		assert.Equal(t, len(nodeClaim.Spec.Requirements), 4, " NodeClaim must have 3 NodeSelector Requirements")
```
Comment (bangqipropel) [inline pkg/utils/nodeclaim/nodeclaim_test.go:208]: I think if you change the previous to 4, I think the sentence should be " NodeClaim must have 3 NodeSelector Requirements".

Which one is the requirement you added?

Comment (smritidahal653) [reply pkg/utils/nodeclaim/nodeclaim_test.go]: Fixed. And the requirement added was karpenter.k8s.aws/instance-gpu-count should be greater than 0 for aws

**Review by Fei-Guo — APPROVED** [review]

File: charts/kaito/workspace/templates/deployment.yaml [hunk 2]
```diff
@@ -47,6 +47,8 @@ spec:
               value: {{ .Values.presetRegistryName }}
             - name: CLOUD_PROVIDER
               value: {{ .Values.cloudProviderName }}
+            - name: CLUSTER_NAME
+              value: {{ .Values.clusterName }}
```
Comment (ishaansehgal99) [inline charts/kaito/workspace/templates/deployment.yaml:51]: consider 
value: {{ .Values.clusterName | default "default-cluster-name" }}

**Review by ishaansehgal99 — APPROVED** [review]

**Review by zhuangqh — APPROVED** [review]

File: charts/kaito/workspace/templates/deployment.yaml (hunk 2 above)
Comment (smritidahal653) [reply charts/kaito/workspace/templates/deployment.yaml:51]: added default to values.yaml

**Review by ishaansehgal99 — APPROVED** [review]

**Pull request merged**